test/functional/http/post/1.second.secret.js

```diff
@@ -0,0 +1,438 @@
+'use strict';
+
+var node = require('../../../node');
+var shared = require('../../shared');
+var constants = require('../../../../helpers/constants');
+
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var sendSignaturePromise = require('../../../common/apiHelpers').sendSignaturePromise;
+
+var onNewBlockPromise = node.Promise.promisify(node.onNewBlock);
+
+describe('POST /api/transactions (type 1) register second secret', function () {
+
+	var badTransactions = [];
+	var goodTransactions = [];
+	var badTransactionsEnforcement = [];
+	var goodTransactionsEnforcement = [];
+	var pendingMultisignatures = [];
+
+	var account = node.randomAccount();
+	var accountEmptySecondPassword = node.randomAccount();
+	accountEmptySecondPassword.secondPassword = '';
+	var accountNoFunds = node.randomAccount();
+	var accountScarceFunds = node.randomAccount();
+	var accountNoSecondPassword = node.randomAccount();
+
+	var transaction, signature;
+
+	// Crediting accounts
+	before(function () {
+		var promises = [];
+		promises.push(creditAccountPromise(account.address, 100000000000));
+		promises.push(creditAccountPromise(accountEmptySecondPassword.address, 100000000000));
+		promises.push(creditAccountPromise(accountScarceFunds.address, constants.fees.secondsignature));
+		promises.push(creditAccountPromise(accountNoSecondPassword.address, constants.fees.secondsignature));
+
+		return node.Promise.all(promises).then(function (results) {
+			results.forEach(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').that.is.not.empty;
+			});
+		}).then(function (res) {
+			return onNewBlockPromise();
+		});
+	});
+
+	describe('schema validations', function () {
+
+		shared.invalidAssets(account, 'signature', badTransactions);
+	});
+
+	describe('transactions processing', function () {
+
+		it('with no funds should fail', function () {
+			transaction = node.lisk.signature.createSignature(accountNoFunds.password, accountNoFunds.secondPassword);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.not.be.ok;
+				node.expect(res).to.have.property('message').to.equal('Account does not have enough LSK: ' + accountNoFunds.address + ' balance: 0');
+				badTransactions.push(transaction);
+			});
+		});
+
+		it('with scarce funds should be ok', function () {
```

Comment:
`using all funds should be ok`

Reply:
When I define the case, I credit the account with `scarce funds` meaning short supply of funds which are going to be spent.